raft/doc.go

```diff
@@ -180,5 +180,89 @@ cannot be removed any more since the cluster cannot make progress.
 For this reason it is highly recommended to use three or more nodes in
 every cluster.
 
+MessageType
+
+Package raft sends and receives message in Protocol Buffer format (defined
+in raftpb package). Each state (follower, candidate, leader) implements its
+own Step method to advance with the given raftpb.Message. And each step is
+determined by its raftpb.MessageType, as follows:
+
+	'MsgHup' is used for election. If a node is follower or candidate, the
+	tick function in 'raft' struct is set as 'tickElection'. If a follower
+	has not received any heartbeat within election timeout, the follower
+	passes 'MsgHup' to its Step method and becomes candidate to start a new
+	election.
+
+	'MsgBeat' is for sending heartbeat. If a node is leader, tick function
+	in 'raft' struct is set as 'tickHeartBeat' that sends periodic heartbeat
+	messages in 'MsgBeat' type to its followers.
```

Comment:
Thanks for clarification. This was confusing to me as well. Will try to clarify more.